### DISCUSSION TITLE
local development requests are not bundled

### DISCUSSION THREAD

**Discussion by acetinick** [ElMassimo/vite_ruby#495] [General]
Is there a reason why for local development my JS imports are not bundled?  In staging working fine, but local dev its requesting hundreds of requests and slows down local dev consdierably.

Localhost
<img width="837" alt="Screenshot 2024-09-13 at 4 57 05 AM" src="https://github.com/user-attachments/assets/9ddd4629-8a12-4daf-844c-beb0b29e2544">

Staging
<img width="670" alt="Screenshot 2024-09-13 at 4 57 51 AM" src="https://github.com/user-attachments/assets/898167d8-9c9e-4894-89d0-006276b7a219">

**Comment by ElMassimo** [comment]
Hi Andrew!

[See the Vite overview, which goes into detail about why this is usually a good thing](https://vitejs.dev/guide/why.html#slow-server-start).

By using dynamic imports and loading only what the current page needs, you avoid having to bundle the entire app, improving startup time, and greatly improving HMR time.

If your app is not taking advantage of page-based splitting and is causing the entire app to be fetched, you can try [skipping the Rails proxy](https://vite-ruby.netlify.app/config/index.html#skipproxy-experimental), and [enabling https](https://vite-ruby.netlify.app/config/index.html#https) so that the browser can leverage [http2](https://vitejs.dev/config/preview-options.html#preview-https) allowing all of those requests to use the same connection.

The reason this is not a default, is that configuring https correctly is not straightforward.
